tests/unit/lms/services/jwt_test.py

```diff
@@ -57,24 +61,90 @@ def test_encode_with_secret(self, svc, jwt):
         )
         assert encoded_jwt == jwt.encode.return_value
 
+    def test_decode_lti_token_with_no_kid(self, svc):
+        jwt_str = self.encode_jwt({"key": "value"}, headers={"key": "value"})
+
+        with pytest.raises(InvalidJWTError) as err_info:
+            svc.decode_lti_token(jwt_str)
+
+        assert "Missing 'kid' value in JWT header" in str(err_info.value)
+
+    def test_decode_lti_token_with_no_registration(self, svc, lti_registration_service):
+        jwt_str = self.encode_jwt({"aud": "AUD", "iss": "ISS"}, headers={"kid": "KID"})
+        lti_registration_service.get.return_value = None
+
+        with pytest.raises(InvalidJWTError) as err_info:
+            svc.decode_lti_token(jwt_str)
+
+        assert "Unknown registration for lti_token. iss:'ISS' aud:'AUD'." in str(
+            err_info.value
+        )
+
+    def test_decode_lti_token_with_invalid_jwt(self, svc, jwt):
+        jwt.decode.side_effect = [{"aud": "AUD", "iss": "ISS"}, InvalidTokenError()]
+        jwt_str = self.encode_jwt({"key": "value"}, headers={"kid": "KID"})
+
+        with pytest.raises(InvalidJWTError):
+            svc.decode_lti_token(jwt_str)
+
+    def test_decode_lti_token(self, svc, jwt, lti_registration_service):
+        jwt.decode.return_value = {"aud": "AUD", "iss": "ISS"}
+
+        payload = svc.decode_lti_token(sentinel.token)
+
+        lti_registration_service.get.assert_called_once_with("ISS", "AUD")
+        jwt.PyJWKClient.assert_called_once_with(
+            lti_registration_service.get.return_value.key_set_url
+        )
+        jwt.decode(
+            sentinel.token,
+            key=jwt.PyJWKClient.return_value.get_signing_key_from_jwt.return_value.key,
+            audience="AUD",
+            algorithms=["RS256"],
+        )
```

Comment:
What does this do? Should this be an `assert_called_once_with()`?

Actually that won't work as there are two calls, you'd have to assert both